src/transaction/components/TransactionButton.tsx

```diff
@@ -59,7 +60,10 @@ export function TransactionButton({
     return buttonText;
   }, [buttonText, errorMessage, receipt]);
 
-  const handleSubmit = useCallback(() => {
+  const handleSubmit = useCallback(async () => {
+    if (!address) {
+      await connectAsync({ connector: connectors[0] });
```

Comment:
Let's avoid create a branch of developer experience by adding a new way to solve the same problem we did with https://github.com/coinbase/onchainkit/pull/1173 .

We want to make sure both experience re-use Wallet logic, and do not create internal branch on how the same problem can be solved.